src/main/java/org/jahia/modules/sitemap/filter/SitemapCacheFilter.java

```diff
@@ -59,126 +54,85 @@
 
     private static final Logger logger = LoggerFactory.getLogger(SitemapCacheFilter.class);
 
+    private static final String PREFIX_CACHE_NAME = "sitemap-cache-";
+    private Ehcache sitemapCache;
+    private SitemapService sitemapService;
+
+    @Reference
+    public void setSitemapService(SitemapService sitemapService) {
+        sitemapCache = sitemapService.getSitemapEhCache();
+        this.sitemapService = sitemapService;
+    }
+
     @Activate
     public void activate() {
-        // Disable filter as it will be rework (and sitemap node not used anymore)
-        setDisabled(true);
         setPriority(15f);
-        setApplyOnNodeTypes("jseont:sitemapResource,jseont:sitemap");
+        setApplyOnNodeTypes("jseomix:sitemap,jseomix:sitemapResource");
         setApplyOnModes("live");
-        setDescription("Filter for creating sitemap file nodes for caching");
+        setDescription("Filter for sitemap caching");
         logger.debug("Activated SitemapCacheFilter");
     }
 
     /**
-     * Serve cache contents if cache exists and is not expired. Otherwise, flush sitemap cache and re-render
+     * Stop render if cache exists and not expired
      */
-    @Override public String prepare(RenderContext renderContext, Resource resource, RenderChain chain) throws Exception {
-        if (!needsCaching(resource)) return null;
-
-        JCRNodeWrapper cacheNode = getCacheNode(resource.getNode());
-        InputStream inputStream = null;
-        try {
-            if (isValidCache(cacheNode)) {
-                inputStream = cacheNode.getFileContent().downloadFile();
-                return IOUtils.toString(inputStream, StandardCharsets.UTF_8);
-            }
-        } catch (IOException e) { // something happened; re-render
-            logger.error("Unable to read sitemap file cache contents.");
-        } finally {
-            IOUtils.closeQuietly(inputStream);
+    @Override
+    public String prepare(
+            RenderContext renderContext,
+            Resource resource,
+            RenderChain chain
+    ) throws Exception {
+
+        if (needsCaching(resource)) return null;
+
+        String targetSitemapCacheKey = PREFIX_CACHE_NAME + resource.getNode().getLanguage();
+
+        if (sitemapCache.get(targetSitemapCacheKey) != null) {
+            return "";
```

Comment:
You can do all the operation in the prepare state and do nothing in the execute, returning the output from the cache will work

Reply:
I made some changes